### PR TITLE
docs(orc8r): Adds documentation to deploy using Juju

### PR DESCRIPTION
**Pull request by gruyaume**
## Summary

Adds documentation to deploy orc8r using Juju.

### CODE REVIEW

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]: Perhaps the items a) - d) here can be simplified by removing the repeated text. Something like:
... a Juju controller on it. Four options for doing this are provided here along with the corresponding guides:
* [Microk8s](https://juju.is/docs/olm/microk8s)
* [Google Cloud - GKE](https://juju.is/docs/olm/google-kubernetes-engine-(gke))
etc.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]:
```suggestion
## 3. Deploy charmed Magma Orchestrator
```

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]:
```suggestion
Deploy Orchestrator:
```

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]:
```suggestion
## 5. Create the Orchestrator admin user
```

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: Agreed, fixed

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (maxhbr) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]: the `(Beta)` is pretty unspecific. Can you explain in more details whether the deployment is `experimental` or `production ready` and whether it is `feature complete`? Maybe also link to the repository where the development of the juju deployment is happening.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (Neudrino) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]: I just want to make sure that the double `..` in front of `data` are not a typo. I have never seen a path as special as this before. :innocent:

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: I added reference to the project's homepage / repository and also mentioned that it is not yet production ready. I don't think I should add more info as to **why** in the how-to guide as the project evolves every week and a how-to guide shouldn't contain such information. But for reference, here are the main reasons:
- It does not include integration with logging and monitoring tools (prometheus, alertmanager, grafana, fluentd, elasticsearch)
- It does not support scaling of components.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: Yeah that's intended :) We will change this soon though but if you are to deploy it today this would be how you retrieve the admin_operator pfx package.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (maxhbr) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: > [...] as the project evolves every week [...]

but it contain a "what to expect right now (at time of writing)". If there are limitations and missing features, they should be listed in the beginning so that one can understand whether this deployment matches the expectations or is suitable for the use-case.

Since it is documentation about another project (https://github.com/canonical/charmed-magma) it might be an alternative to have just a "what to expect and where to find documentation" that points to somewhere else.

EDIT: just saw that you added `Charmed-Magma is in Beta and is not yet production ready.`. Not sure if that is sufficient. I would like to avoid this potentially frustrating journey, where someone installs a charmed magma just to find out that one specific part is missing.

Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: I want to stress the fact that this new document is a how-to guide, not a reference document. I don't want to bundle different types of documentations into one place just because they relate to a same subject. If we need reference documentation on what is this project, what state it is in, etc. I don't mind creating a new document for this but I don't think this information should live in a how-to guide.

On the subject of simply pointing to another source of documentation, I would disagree. This repo here contains Magma's official documentation and through this new how-to guide, an operator would be deploying Magma. Even though operators are handled in another project, the software deployed is the same: The OCI images and go code that we are deploying are identical. Having different places where one would land to deploy the same software would increase confusion in my opinion.

Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: And for reference, here are the principles I followed to base my views on how-to guides: https://diataxis.fr/how-to-guides/

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [inline docs/readmes/orc8r/how_to_deploy_using_juju.md]: For consistency with the rest of the docs, I would perhaps remove the "How-To" here, since many other pages fall under this umbrella but don't have this in the name.

Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: That's a fair comment, the problem (I think) is that we have some pages that are "How-to guides", "Tutorials", "Reference", and "Explanation" but those aren't categorized so it's a bit of a mix bag. Here I wanted to make it obvious that this page was a "How-to Guide".

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: @maxhbr Can you please review my comments and either resolve the thread or provide direction?

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: @voisey What do you think?

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (maxhbr) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: still think that it should highlight that there are known limitations.

If I find that document as is right now I would expect it to be a fully working and complete deployment. But as I understood that is not yet the case.

Please correct me if I am wrong here.

maybe:
```diff 
- > Charmed-Magma is in Beta and is not yet production ready.
+ > Charmed-Magma is in Beta and is not yet production ready or feature complete.
```
that should be a sufficient pointer.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: Apologies for the delay, I was away for a few days.

Sure, that makes sense. Personally, I would prefer to it to be removed for now and for it to be part of a general restructuring, if that's considered desirable by other people too, just so we don't have inconsistencies that are potentially confusing. I don't see it as a blocker though.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: Understood, I just applied your recommendation.

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (gruyaume) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: Understood, I just applied your recommendation.

**Review by maxhbr — APPROVED** [review]

File: docs/readmes/orc8r/how_to_deploy_using_juju.md
Comment (voisey) [reply docs/readmes/orc8r/how_to_deploy_using_juju.md]: Thanks!

**Review by voisey — APPROVED** [review]

**Comment by voisey** [comment]
@gruyaume I think this is good to go once the two CI failures are addressed. Could you please take a look at them? It looks like 'DCO check' is failing because some of the commits aren't signed and 'Markdown lint check' because a change to a generated file is missing - I think you can run `make -C magma/docs` to generate the change, which can then be pushed. Let me know if any of this is unclear!

**Comment by gruyaume** [comment]
@voisey @maxhbr Is there anything I can do at this point to get this merged?

**Comment by maxhbr** [comment]
I think you have the necessary approvals and reviews and I see no blocking comment -> `ready for merge`-label

**Comment by voisey** [comment]
@gruyaume It seems you still need to sign your commits. Once that's done, we can merge it

**Comment by maxhbr** [comment]
maybe sqaush and push again?

**Comment by maxhbr** [comment]
DCO Results:

```
  d4da9647a62f0059a9685c31c68e84b7d99e3119    The sign-off is missing.
  c3b402006f681ee8ed84c362e9f2144c5f779046    The sign-off is missing.
  0ab67432fc6458662186a0d7e2ee05b8086e954c    The sign-off is missing.
  48642c43357c9adc4124f0331b78ba1937a44d92    Expected "guillaume <guillaume.belanger27@gmail.com>", but got "Guillaume Belanger <guillaume.belanger@canonical.com>".
  e71412b3876b1f761e78ec0dd7dea9c78cda2304    The sign-off is missing.
  7bd9aefdaa7055fc1fb2d8ade69831aefaae6447    Expected "guillaume <guillaume.belanger27@gmail.com>", but got "Guillaume Belanger <guillaume.belanger@canonical.com>".
  2b13ea1b2d41ceaf649bce6187e133eaf2008f57    The sign-off is missing.
  fb9083a17e0b2898c77742f1b262c46eed759bbd    The sign-off is missing.
  eabf1dd44a5a08a9eff10f7781fc869afa02e5b5    The sign-off is missing.
  47383dba9b9fe0d7910ac93a2298642215d84b7a    The sign-off is missing.
  47eeeee640f3db18bb3af716032d31d06173662c    The sign-off is missing.
  d0a257e76837806b62d2c1467eebce092b56f30b    The sign-off is missing.
  0a044ba64cfa3ce255582b9a4ce1e73e3efb5b72    The sign-off is missing.
```

**Comment by maxhbr** [comment]
you have merge conflicts

**Pull request closed without merging**
